### DISCUSSION TITLE
`sx` types added to Transition from `@headlessui/react` make props type too large to represent in TypeScript

### DISCUSSION THREAD

**Discussion by RishikeshDarandale** [system-ui/theme-ui#1781] [Q&A]
Using `headlessui/react` library components like `Dialog`, getting an `ts` error as `Expression produces a union type that is too complex to represent.ts(2590)`

```
/** @jsx jsx */
import { Dialog, Transition } from '@headlessui/react';
import { FunctionComponent, } from 'react';
import { Box, jsx } from 'theme-ui';

interface Props {
};

const Abcd: FunctionComponent<Props> = (
  props: Props
) => {
   return (
    <Box>
      <Transition>
      </Transition>
    </Box>
  );
};

```

Anybody knows how to resolve this error?

**Comment by hasparus** [comment]
Are you getting it on Box?

**Reply by RishikeshDarandale** [reply]
It's happening on `Transition` / `Dialog` component of headlessui. Let me create a sample project.

**Reply by RishikeshDarandale** [reply]
Created a [repository](https://github.com/RishikeshDarandale/theme-ui-error-with-headlessui) to showcase the issue and issue is documented in [README](https://github.com/RishikeshDarandale/theme-ui-error-with-headlessui/blob/main/README.md) as well.

**Reply by hasparus** [reply]
Sadlly, I don't think there are any satisfying solutions for `ts(2590)`. Types for `sx` are quite large, an types for Headless UI components are also pretty complex. 👇

![image](https://user-images.githubusercontent.com/15332326/120039171-c310ea80-c004-11eb-8f6f-b1aeb134e497.png)

You can work around it with `as any` or use Transition and Dialog in files where TypeScript doesn't know about the pragma.

**Reply by RishikeshDarandale** [reply]
Thanks @hasparus . Yes, I saw that there is no solution to typescript error. I created this one to check if there are better work around.

Can you please elaborate more on using `any`?

If I use, those components without pragma in the file means I need to style them using `classNames`. Are there any guideline from `theme-ui` to style the component without `sx` and `css` props?

Is there any roadmap for `theme-ui` to add more components like `dialog`, etc?
